File: exercism/python/difference-of-squares/difference_of_squares.py

def difference(n):
    return square_of_sum(n) - sum_of_squares(n)


def square_of_sum(n):
    return sum(range(n + 1)) ** 2


def sum_of_squares(n):
    return sum(m ** 2 for m in range(n + 1))

File: exercism/javascript/difference-of-squares/difference-of-squares.js
const fp = require('lodash/fp')

const square = (n) => n * n

class Squares {
  constructor (n) {
    this.n = n
    this.sumOfSquares = fp.sum(fp.range(1, n + 1).map(square))
    this.squareOfSums = square(fp.sum(fp.range(1, n + 1)))
    this.difference = this.squareOfSums - this.sumOfSquares
  }
}

module.exports = Squares
